driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -442,12 +442,13 @@ public NumberExpression isNumberOr(final NumberExpression other) {
     }
 
     public BooleanExpression isInteger() {
-        return this.isNumber().cond(this.eq(this.round()), of(false));
+        return this.isNumber().and(this.eq(this.round()));
     }
 
     @Override
     public IntegerExpression isIntegerOr(final IntegerExpression other) {
-        return this.isInteger().cond(this, other);
+        return this.isNumber().and(this.eq(this.round()))
```

Comment:
It appears that this method still can be implemented as `return isInteger().cond(this, other);`, I don't think a change is needed here.

Reply:
Used switch.

Reply:
I meant that `isIntegerOr` could still be implemented via `isInteger`. But that no longer applies.